plugin-oar2/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+                let job_separated = job_name.split_once("_");
+                let job_id = job_separated.context("Invalid oar cgroup.")?.1.parse()?;
+
+                let cpu_job_path = cgroup_cpu_path.join(&job_name);
+                let memory_job_path = cgroup_memory_path.join(&job_name);
+
+                let cpu_file_path = cpu_job_path.join("cpuacct.usage");
+                let memory_file_path = memory_job_path.join("memory.usage_in_bytes");
+
+                let cgroup_cpu_file = File::open(&cpu_file_path)
+                    .with_context(|| format!("Failed to open CPU usage file at {}", cpu_file_path.display()))?;
+                let cgroup_memory_file = File::open(&memory_file_path)
+                    .with_context(|| format!("Failed to open memory usage file at {}", memory_file_path.display()))?;
+
+                let initial_source = Box::new(OarJobSource {
+                    cpu_metric,
+                    memory_metric,
+                    cgroup_cpu_file,
+                    cgroup_memory_file,
+                    cpu_file_path,
+                    memory_file_path,
+                    job_id,
+                });
+
+                alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+            }
+        }
+        Ok(())
+    }
+
+    fn post_pipeline_start(&mut self, pipeline: &mut alumet::pipeline::runtime::RunningPipeline) -> anyhow::Result<()> {
+        let control_handle = pipeline.control_handle();
+        let config_path = self.config.path.clone();
+        let plugin_name = self.name().to_owned();
+
+        let metrics = self.metrics.clone().unwrap();
+        let cpu_metric = metrics.cpu_metric;
+        let memory_metric = metrics.memory_metric;
+        let poll_interval = self.config.poll_interval;
+
+        struct JobDetector {
+            config_path: PathBuf,
+            cpu_metric: TypedMetricId<u64>,
+            memory_metric: TypedMetricId<u64>,
+            control_handle: ControlHandle,
+            plugin_name: String,
+            poll_interval: Duration,
+        }
+
+        impl EventHandler for JobDetector {
+            fn handle_event(&mut self, event: Result<Event, notify::Error>) {
+                log::debug!("Handle event function");
+                if let Ok(Event {
+                    kind: EventKind::Create(_),
+                    paths,
+                    ..
+                }) = event
+                {
+                    log::debug!("Paths: {paths:?}");
+                    for path in paths {
+                        if let Some(job_name) = path.file_name() {
+                            let job_name = job_name.to_str().unwrap();
+                            if job_name.chars().any(|c| c.is_numeric()) {
+                                let job_separated = job_name.split_once("_");
+                                let job_id = job_separated.context("Invalid oar cgroup").unwrap().1.parse().unwrap();
```

Comment:
I don't know what's best to manage these two unwrap, but the plugin should not panic on that. It could log an error and skip the current job or set a default value I don't know

Reply:
Creating a separate function that returns a result would be the best way to do it. It would allow to use `?`.